### PR TITLE
tailscale: use V2 client for Webhooks

### PR DESCRIPTION
**Pull request by oxtoacart**
Updates tailscale/corp#21867

Depends on https://github.com/tailscale/tailscale-client-go/pull/91

### CODE REVIEW

**Comment by oxtoacart** [comment]
Acceptance test passed - https://github.com/tailscale/corp/actions/runs/10225845945/job/28295382372

**Comment by oxtoacart** [comment]
Acceptance test passed.

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	github.com/tailscale/hujson v0.0.0-20221223112325-20486734a56a
 	github.com/tailscale/tailscale-client-go v1.17.1-0.20240729175651-90a1e935cc19
-	github.com/tailscale/tailscale-client-go/v2 v2.0.0-20240802213233-7518e6604513
+	github.com/tailscale/tailscale-client-go/v2 v2.0.0-20240805140419-bbf53147c21e
```
Comment (mpminardi) [inline go.mod:13]: Reminder to update to ref from main before merging

**Comment by oxtoacart** [comment]
Acceptance passed after rebase, merging.

**Pull request merged**